File: javascript/InsertionSort.js
function InsertionSort(a){
    for (let j = 1; j < a.length; j++){
        let temp = a[j];

        let i = j - 1;

        while ( i >= 0 && temp < a[i]){
            a[i + 1] = a[i];
            i--;
        }
        a[i + 1] = temp;
        

        }

    }
function PrintArr(a){
        console.log(a);
    
}

let a = [70, 15, 2, 51, 60];

console.log("Antes de ordenar los elementos del arreglo: ");
PrintArr(a);

InsertionSort(a);

console.log("Despues de ordenar los elementos del arreglo");
PrintArr(a);

File: python/InsertionSort.py
def InsertionSort(a):
    for i in range (1, len(a)):
        temp = a[i]
        
        j = i - 1
        
        while j >= 0 and temp < a[j]: 
            a[j + 1] = a[j]
            j = j-1 
        a[j + 1] = temp
def printArr(a): 
    for i in range (len(a)):
        print (a[i], end = ' ')
    print()

a = [70, 15, 2, 51, 60]

print("Antes de ordenar los elementos del arreglo:")
printArr(a)
InsertionSort(a)
print("Despues de ordenar los elementos del arreglo:")
printArr(a)   